android/src/main/java/com/vydia/RNUploader/UploaderModule.kt

```diff
@@ -134,38 +65,24 @@ class UploaderModule(val reactContext: ReactApplicationContext) :
   /**
    * @return whether the upload was started
    */
-  private fun maybeStartUpload(upload: Upload) {
-    if (upload.wifiOnly && wifiOnlyHttpStack == null) return
-    if (!upload.wifiOnly && httpStack == null) return
-
-    val notificationManager =
-      (reactContext.getSystemService(Context.NOTIFICATION_SERVICE) as NotificationManager)
-    initializeNotificationChannel(upload.notificationChannel, notificationManager)
-
-    val requestId = UUID.randomUUID().toString()
-
-    val request = if (upload.requestType == Upload.RequestType.RAW) {
-      UploadRequestBinary(reactContext, upload.url, upload.wifiOnly).apply {
-        setFileToUpload(upload.path)
-      }
-    } else {
-      UploadRequestMultipart(reactContext, upload.url, upload.wifiOnly).apply {
-        addFileToUpload(upload.path, upload.field)
-        upload.parameters.forEach { (key, value) -> addParameter(key, value) }
-      }
-    }
-    Log.i(TAG, "starting request ID $requestId for ${upload.id}")
-
-    request.apply {
-      setMethod(upload.method)
-      setMaxRetries(upload.maxRetries)
-      setUploadID(requestId)
-      upload.notification.let { if (it != null) setNotificationConfig { _, _ -> it } }
-      upload.headers.forEach { (key, value) -> addHeader(key, value) }
-      startUpload()
-    }
-
-    upload.requestId = UploadServiceId(requestId)
+  private fun startUpload(options: ReadableMap): String {
+    val upload = Upload.fromOptions(options)
+    val data = Gson().toJson(upload)
+    val constraints = Constraints.Builder().setRequiredNetworkType(NetworkType.CONNECTED)
+    if (upload.wifiOnly) constraints.setRequiredNetworkType(NetworkType.UNMETERED)
```

Comment:
a complex system to enable wifiOnly behaviors has now been shrunk down to these 2 lines